### ISSUE TITLE
React has detected a change in the order of Hooks called by Root

### ISSUE THREAD

**Issue by xiaoxiaosaohuo** [preactjs/signals#180]
[](url)
<img width="1028" alt="Screenshot 2022-09-19 at 2 31 17 PM" src="https://user-images.githubusercontent.com/15315816/190961011-d64ceaeb-8447-46d8-bcb6-a0d595a8f3b1.png">

I'm using Next.js, when I navigated to another page, this error occurred

this is the target page code

```
const count = signal(0);

function Counter() {
  return <button onClick={() => count.value++}>Value: {count.value}</button>;
}
export default Counter;
```
- next: 12.2.5
- react: 18.2.0
- signals-react: 1.0.2

**Comment by sbesh91** [comment]
I am also seeing a similar error in the console when we have signals on the page.  
<img width="806" alt="image" src="https://user-images.githubusercontent.com/3409138/193656539-7074dedb-6435-4ab9-8063-dff32b3e8b8a.png">
Oddly enough, signals aren't used on the component triggering this error.

**Comment by jmeistrich** [comment]
For what it's worth, I've come to the conclusion that this method of injecting useReducer into the current dispatcher can't possibly work reliably and I've migrated legend-state away from it. We were seeing this bug in Next.js and users saw a similar bug in Expo Router, and in some Framer-Motion animations it was crashing as elements unmounted. After a lot of debugging I was not able to find a reliable solution. That combined with the React team telling us not to do it and that it would break soon made it seem untenable.

I hope you're able to find a solution! But if not I hope I can save you some time knowing that it didn't work for us.

**Comment by marvinhagemeister** [comment]
Fixed by #219
